src/rgw/rgw_rados.cc

```diff
@@ -6009,7 +6009,11 @@ int RGWRados::Bucket::UpdateIndex::guard_reshard(BucketShard **pbs, std::functio
   for (int i = 0; i < NUM_RESHARD_RETRIES; ++i) {
     int ret = get_bucket_shard(&bs);
     if (ret < 0) {
-      ldout(store->ctx(), 5) << "failed to get BucketShard object: ret=" << ret << dendl;
+      if (ret == -ENOENT) {
+        ldout(store->ctx(), 5) << "failed to get BucketShard object: ret=" << ret << ", modifying to " << -ERR_INTERNAL_ERROR << dendl;
+        ret = -ERR_INTERNAL_ERROR;
```

Comment:
change on this path seems sane--nothing has I think happened, so request seems definitely retryable